.github/workflows/pull-requests.yaml

```diff
@@ -65,6 +65,14 @@ jobs:
           GH_TOKEN: ${{ github.token }}
         run: |
           ./tools/get-last-successful-workflow-run-for-artifacts.sh --repo ${{ github.repository }}
+          
+      - name: Turn script into an executable
+        working-directory: ./tools
+        run: chmod +x detect-secrets.sh
+    
+      - name: Run the detect secrets script
+        working-directory: ./tools
+        run: ./detect-secrets.sh
```

Comment:
Please can you create a new job called `detect-secrets` at the top of the workflow and put these two steps in there. The job `end-pull-request-build` will need `detect-secrets` added to its "needs" section - so the pull request cannot end successfully unless that job was successful.